pkg/resources/statefulsets/statefulset.go

```diff
@@ -429,74 +429,48 @@ func newStatefulSetForCR(cr *brokerv1alpha1.ActiveMQArtemis) *appsv1.StatefulSet
 	return ss
 }
 
-type ErrMissingField string
-
-func (e ErrMissingField) Error() string {
-
-	return string(e) + " is Required Field"
+func checkSSLEnabled(cr *brokerv1alpha1.ActiveMQArtemis) bool {
+	reqLogger := log.WithName(cr.Name)
+	var sslEnabled = false
+	if len(cr.Spec.SSLConfig.SecretName) != 0 && len(cr.Spec.SSLConfig.KeyStorePassword) != 0 && len(cr.Spec.SSLConfig.KeystoreFilename) != 0 && len(cr.Spec.SSLConfig.TrustStorePassword) != 0 && len(cr.Spec.SSLConfig.TrustStoreFilename) != 0 {
```

Comment:
So a complete SSLConfig block results in proper SSL configuration, and its omission leads to a configuration without SSL. That's good so far.

If you see a user provide a CR with an SSLConfig containing SecretName, KeystoreFilename and KeyStorePassword, for example, what would you assume their intentions are? As I understand, the code will assume they do not intend to configure SSL and will happily deploy an environment without it. That's not how I would interpret their motives. I think incomplete configuration should result in a failed deployment with proper feedback.

Reply:
CRD validation already in place in case if user intention to create SSL configuration has to provide all properties which marked as required, otherwise the CR will stop ever being created in the first place.
https://github.com/rh-messaging/activemq-artemis-operator/blob/d0efe77e026e3a91d0483a0713d3547600537252/deploy/crds/broker_v1alpha1_activemqartemis_crd.yaml#L43